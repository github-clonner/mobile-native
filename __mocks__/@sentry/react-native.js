export const captureMessage = jest.fn();
export default {};
