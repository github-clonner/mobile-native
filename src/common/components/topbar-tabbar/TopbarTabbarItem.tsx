import React, { Component } from 'react';

import { inject, observer } from 'mobx-react';

import {
  Text,
  TouchableOpacity,
  StyleSheet,
  View,
  StyleProp,
  ViewStyle,
  GestureResponderEvent,
} from 'react-native';

import { CommonStyle } from '../../../styles/Common';
import colors from '../../../styles/Colors';
import ThemedStyles from '../../../styles/ThemedStyles';
import { useTopbarBarStore } from './TopbarTabbarContext';

interface Props {
  id: string;
  children: string;
  //onPress: (e: GestureResponderEvent) => {};
  style?: StyleProp<ViewStyle>;
}

/**
 * Topbar Tabbar Item
 */
export const TopbarTabbarItem = observer((props: Props) => {
  const store = useTopbarBarStore();

  const isSelected = (): ViewStyle | null => {
    // const activeTabId = store.activeTabId;
    // return activeTabId == txt ? styles.tabSelected : null;
    return store.activeTabId === props.id ? styles.tabSelected : null;
  };

  const onPress = (e: GestureResponderEvent) => {
    store.setActiveTabId(props.id);
  };

  return (
    <TouchableOpacity
      style={[
        styles.tab,
        {
          borderBottomColor: ThemedStyles.getColor('secondary_background'),
        },
        isSelected(),
      ]}
      onPress={onPress}>
      <Text style={CommonStyle.fontM}>{props.children}</Text>
    </TouchableOpacity>
  );
});

export default TopbarTabbarItem;

//TODO: move to common style
const styles = StyleSheet.create({
  tab: {
    justifyContent: 'center',
    alignItems: 'center',
    borderBottomWidth: 4,
    paddingHorizontal: 0,
    marginRight: 20,
  },
  tabSelected: {
    borderBottomColor: colors.primary,
  },
});
