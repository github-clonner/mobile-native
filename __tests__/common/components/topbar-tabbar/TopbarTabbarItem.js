import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { shallow, render } from 'enzyme';
import TopbarTabbarItem from '../../../../src/common/components/topbar-tabbar/TopbarTabbarItem';

// Note: test renderer must be required after react-native.
import renderer, { act } from 'react-test-renderer';

it('renders correctly', () => {
  const topbarTabbar = renderer
    .create(<TopbarTabbarItem id="first">First Label</TopbarTabbarItem>)
    .toJSON();
  expect(topbarTabbar).toMatchSnapshot();
});

it('should have label rendered', () => {
  const topbarTabbarItem = shallow(
    <TopbarTabbarItem id="first">First Label</TopbarTabbarItem>,
  );

  expect(topbarTabbarItem.find(Text).props().children).toMatch('First Label');
});

it('should update our store on press', () => {
  const topbarTabbarItem = shallow(
    <TopbarTabbarItem id="first">First Label</TopbarTabbarItem>,
  );

  act(() => {
    topbarTabbarItem.simulate('press');
  });

  // TODO: figure out how to mock the store properly
});
