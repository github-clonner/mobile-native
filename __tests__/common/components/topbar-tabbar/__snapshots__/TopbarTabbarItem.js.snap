// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#202527",
      "borderBottomWidth": 4,
      "justifyContent": "center",
      "marginRight": 20,
      "opacity": 1,
      "paddingHorizontal": 0,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 14,
      }
    }
  >
    First Label
  </Text>
</View>
`;
